MediaProcessor/src/FFmpegCommandBuilder.h

```diff
@@ -0,0 +1,111 @@
+#ifndef FFMPEGCOMMANDBUILDER_H
+#define FFMPEGCOMMANDBUILDER_H
+
+#include <filesystem>
+
+#include "CommandBuilder.h"
+#include "FFmpegSettingsManager.h"
+
+namespace fs = std::filesystem;
+
+namespace MediaProcessor {
+
+/**
+ * @brief Builds individual FFmpeg commands
+ */
+class FFmpegCommandBuilder : public CommandBuilder {
+   public:
+    /**
+     * @brief Initializes the FFmpegCommandBuilder and adds the FFmpeg binary
+     *        path as the first argument
+     *
+     * @param ffmpegSettings A reference to an instance of FFmpegSettingsManager
+     */
+    FFmpegCommandBuilder(FFmpegSettingsManager& ffmpegSettings);
+
+    /**
+     * @brief Adds overwrite flag
+     *
+     * @return A reference to the current object for method chaining
+     */
+    FFmpegCommandBuilder& addOverwrite();
+
+    /**
+     * @brief Adds input flag and input file path
+     *
+     * @param inputFile The input file path
```

Comment:
For simple variables like these, we don't need to add a `@param` entry. For large headers it can introduce some clutter.